parcels/spatialhash.py

```diff
@@ -242,69 +305,107 @@ def _barycentric_coordinates(nodes, point, min_area=1e-8):
     return barycentric_coords
 
 
-def _planar_quad_area(lat, lon):
-    """Computes the area of each quadrilateral face in a curvilinear grid.
-    The lon and lat arrays are assumed to be 2D arrays of points with dimensions (n_y, n_x).
-    The area is computed using the Shoelace formula.
-    This method is only used during hashgrid construction to determine the size of the hash cells.
+def _latlon_rad_to_xyz(lat, lon):
+    """Converts Spherical latitude and longitude coordinates into Cartesian x,
+    y, z coordinates.
+    """
+    x = np.cos(lon) * np.cos(lat)
+    y = np.sin(lon) * np.cos(lat)
+    z = np.sin(lat)
```

Comment:
Are you sure this is the right formula? Eqs 4-6 on https://mathworld.wolfram.com/SphericalCoordinates.html have some of the `sin` and `cos` swapped?